aws-devicefarm-gradle-plugin/src/main/groovy/com.amazonaws.devicefarm/reports/TestResultJUnitXMLReporter.java

```diff
@@ -0,0 +1,145 @@
+import com.amazonaws.services.devicefarm.model.ListTestsRequest;
+import com.amazonaws.services.devicefarm.model.Test;
+import com.amazonaws.services.devicefarm.model.Counters;
+
+import org.gradle.api.logging.Logger;
+
+public class TestResultJUnitXMLReporter
+{
+    private Logger logger;
+    private AWSDeviceFarm api;
+    private DeviceFarmUtils utils;
+    private Run run;
+    private String junitXmlStr = "<?xml version=\"1.0\" ?>";
+    
+    public TestResultJUnitXMLReporter(Run run,
+                                      DeviceFarmExtension extension,
+                                      Logger logger,
+                                      AWSDeviceFarm deviceFarmClient,
+                                      DeviceFarmUtils utils) {
+        this.logger = logger;
+        this.api = deviceFarmClient;
+        this.utils = utils;
+        this.run = run;
+    }
+    
+    private List<Job> getJobs(Run run) {
+        List<Job> jobs = api.listJobs(new ListJobsRequest().withArn(run.getArn())).getJobs();
+        logger.info("Found " + new Integer(jobs.size()).toString() + " jobs for run with url: " + utils.getRunUrlFromArn(run.getArn()));
+        for (int jobIndex = 0; jobIndex < jobs.size(); jobIndex++) {
+            Job curJob = jobs.get(jobIndex);
+            logger.info( "(Type: " + curJob.getType() + ")"
+                         + curJob.getName() + ": "
+                         + curJob.getMessage() + " - "
+                         + curJob.getResult());
+        }
+        
+        return jobs;
+    }
+    
+    public void writeResults(String destination) throws Exception {
+
+        List<Suite> suites = getSuites(getJobs(this.run).get(0));
+        ArrayList<JUnitSuite> junitSuites = new ArrayList<JUnitSuite>();
+
+        for (int suiteIndex = 0; suiteIndex < suites.size(); suiteIndex++) {
+            junitSuites.add(buildTestSuite(suites.get(suiteIndex)));
+        }
+
+        for (int junitSuiteIndex = 0; junitSuiteIndex < junitSuites.size(); junitSuiteIndex++) {
+            JUnitSuite suite = junitSuites.get(junitSuiteIndex);
+
+            if (destination != null) {
+                logger.info("Found JUnitXMLReporter Destination Path: " + destination);
+
+                File testsDirectory = new File(destination).getAbsoluteFile();
+                logger.debug("Resolving to: " + testsDirectory.getAbsolutePath());
+            
+                try {
+                
+                    if (!testsDirectory.exists()) {
+                        logger.info("Did not exist, creating new directory: " + testsDirectory.getAbsolutePath());
+                        testsDirectory.mkdirs();
+                    }
+                
+                    if (testsDirectory.exists()) {
+                        
+                        String xmlFilePath = new File(destination + File.separator + "TEST-" + suite.getClassName() + ".xml").getAbsoluteFile().getAbsolutePath();
```

Comment:
To get test result from multiple devices, this method (`writeResults()`) can be edited, so that this method is done on every `job` available.
```
List<Job> jobs = getJobs(this.run);
for (Job job : jobs) {
    List<Suite> suites = getSuites(job);
    ...
}
```
while for the `xmlFilePath`, we can write it this way
```
String xmlFilePath = new File(destination + File.separator + "TEST-" + job.getDevice().getArn() + "-" + job.getDevice().getName() + "-" + suite.getClassName() + ".xml").getAbsoluteFile().getAbsolutePath();
```
to discriminate test results accross devices. Then we will get results for tests on multiple devices.